### PR TITLE
feat(*): add support for test-failure hook

### LINKED ISSUE #1927: add hook for asserting test failure
Currently you can define tests on your charts that assert success. We also want to be able to assert that a test will fail. For that reason, we need a hook type called test-failure to compliment the current hook `test-success`. You should be able to define a test that should fail and throw this hook annotation on it. When running tests, if this test fails, that would indicate a successful test result result.

### PR DESCRIPTION
**Pull request by michelleN**
* resolves #1927

### CODE REVIEW

**Comment by michelleN** [comment]
cc/ @vdice @thomastaylor312 @technosophos

File: pkg/proto/hapi/chart/metadata.pb.go [hunk 1]
```diff
@@ -74,7 +74,7 @@ type Metadata struct {
 	// The condition to check to enable chart
 	Condition string `protobuf:"bytes,11,opt,name=condition" json:"condition,omitempty"`
 	// The tags to check to enable chart
-	Tags []string `protobuf:"bytes,12,opt,name=tags" json:"tags,omitempty"`
```
Comment (technosophos) [inline pkg/proto/hapi/chart/metadata.pb.go]: How did this change? I am pretty sure this would break stuff.

Comment (michelleN) [inline pkg/proto/hapi/chart/metadata.pb.go]: It was a change that resulted from running `make protoc` after I upgraded to 3.2. 
I took it out of this commit though. Thanks.

Comment (technosophos) [inline pkg/proto/hapi/chart/metadata.pb.go]: Oh. That's weird. I'll have to dig into that.

**Comment by technosophos** [comment]
Waiting to merge until @vdice gives the 👍

**Comment by vdice** [comment]
LGTM -- provides ability to specify a test _expected_ to fail.  Fetched this change locally and tested with the following service test in the `docs/examples/nginx` example chart:
```
apiVersion: v1
kind: Pod
metadata:
  name: "{{.Release.Name}}-service-failing-test"
  annotations:
    "helm.sh/hook": test-failure
spec:
  containers:
    - name: curl
      image: radial/busyboxplus:curl
      command: ['curl']
      args:  [ '{{ template "fullname" .}}-foo:{{default 80 .Values.httpPort}}' ]
  restartPolicy: Never
```
Resulting in
```
$ bin/helm test foolhardy-squirrel
RUNNING: foolhardy-squirrel-service-failing-test
PASSED: foolhardy-squirrel-service-failing-test
```

**Comment by technosophos** [comment]
Merging on behalf of @michelleN since she is out today.

**Pull request merged**